app/src/main/java/com/sdp13epfl2021/projmag/database/UserProfileDatabase.kt

```diff
@@ -55,74 +61,67 @@ class UserProfileDatabase(
                     onFailure(it)
                 }
         } else {
-            Log.d(TAG, "Unable to get the uid from firebase")
+          onFailure(Exception("user id is null"))
         }
     }
 
-    public fun getProfile(onSuccess: (profile: ImmutableProfile?) -> Unit) {
+    /**
+     * Gets the user profile from firebase if it exists
+     * @param onSuccess the closure that's when a profile is downloaded successfully with the fetched profile passed to it
+     * @param onFailure the closure that's called if the download fails
+     */
+    public fun getProfile(
+        onSuccess: (profile: ImmutableProfile?) -> Unit,
+        onFailure: (Exception) -> kotlin.Unit
+    ) {
         val userUid = getUser()?.uid
         if (userUid != null) {
             val docRef = firestore.collection(ROOT).document(userUid)
             docRef.get()
                 .addOnSuccessListener { document ->
                     if (document != null) {
-                        val firstName = document["firstName"] as? String
-                        val lastName = document["lastName"] as? String
-                        val ageLong = document["age"]
+                        val firstName = (document["firstName"] as? String)
+                        val lastName = (document["lastName"] as? String)
+                        val age = (document["age"] as? Long)?.toInt()
+                        val sciper = (document["sciper"] as? Long)?.toInt()
+                        val phoneNumber = (document["phoneNumber"] as? String)
 
-                        val gender = when (document["gender"] as? String) {
+                        val gender = when ((document["gender"] as? String)) {
```

Comment:
why the parenthesis

Reply:
Indeed there is no need for them I will remove them